### DISCUSSION TITLE
Can we recommend a hardware configuration?

### DISCUSSION THREAD

**Discussion by de-authority** [filecoin-project/lotus#6695] [Storage Provider] [answered]
I still see many people asking such questions. Of course, due to the influence of hardware supply, there may not be the best configuration. Can we give a relatively balanced configuration and keep it updated

**Comment by stuberman** [comment, marked as answer]
There are very many options for hardware depending on your goals. What is the best truck? A small pickup truck or a very large semi-truck?

See examples here

https://github.com/filecoin-project/lotus/discussions/6071

**Reply by de-authority** [reply]
Yeah. i saw that. i think the the goal  is not higher than 5pb cluster without modify the master code. too many options make our miner confuse sometimes.

**Reply by stuberman** [reply]
The faster you want to seal 5PiB the more expensive the hardware. You can use an entry level machine but it would take 55 years. You could also build a system to seal that in a week and it would cost a fortune. Right now you are basically asking  x*y = 5 PiB, solve for x and y.

**Reply by jburnett** [reply]
Agree with @stuberman - too many options in terms of tech and end goal. Seems like only configs by tier would be useful, but small, medium, large still involves myriad options. HTH

**Reply by stuberman** [reply]
I think you start with, my goal is to seal 5PiB in nnn days, then calculate the size of the miner capacity to seal per day.

For instance my miner (a Threadripper and Ryzen 9) can seal around 768GiB per day. (18 years for 5 PiB)
You had better consider collateral FIL cost, right now around 7,000 FIL per PiB (or $400K)
Will you be making deals (income possibility) or just pledging and hoping for rewards?

**Reply by jburnett** [reply]
Here's a table we used to get a sense how sectors/day translates to total/year.  To achieve 1 PiB/year requires about 128 32-GiB sectors/day without loss.

[Filecoin Production Rates by 32-GiB Sectors](https://github.com/GranaryTree/community/blob/cc74dfef3d94a92413dada7e4564f1d4574c858f/misc/Filecoin%20Production%20Rates%20by%2032-GiB%20Sectors.png)

Doesn't provide any hardware specs or recommendations, but is a reasonable starting point for understanding scale.

**Reply by de-authority** [reply]
@stuberman Yeah, right.So we can tell them the hardware configuration and sealing speed,which hardware is cost-effective and easy to obtain, and they can set their own goals and sealing time. Of course, collateral FIL also needs to be considered

**Reply by stuberman** [reply]
"Cost effective" is directly related to goals such as speed of sealing and type of sealing.
For instance I might only accept verified deals with 10x power and 10x collateral.
I could seal 1 PiB in 4-5 months in this case although the collateral would be 100K FIL (1/2 million USD?)

**Reply by JelloHuang** [reply]
pls help to provide the different machine and performance. I think it is a good job

**Comment by de-authority** [comment]
Although the hardware configuration is no longer a secret, why miners are still confused? One of the main reasons is that many miners larger than 10PB are running modified code, and the efficiency of the same hardware will be completely different when small miners are running

**Reply by stuberman** [reply]
Many of the big miners are not accepting deals from the public, so they optimize for pledges to make block rewards, short term gain and not a long term vision.

**Reply by de-authority** [reply]
@stuberman  I can't agree more. So we can provide recommend hardware configuration for those miners who do not modify the code. They may be small miners, but they may have long-term vision

**Reply by JelloHuang** [reply]
there r lots of small miners who donot have strong skill in china.So we need help to provide more support. and it is more convenient

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
the minimal requirements are posted in the docs. from there everything is up for discussion. some say you can do it with $5k hardware others wouldn't do it under $10k.

there simply is no optimal hardware for this.

**Reply by de-authority** [reply]
Yeah. Of course, there is no optimal hardware. maybe we can list the performance of some hardware  configurations as a reference, let them have a choice, which is better than the current situation

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
you might want to start a discussion thread for people to post some benchmarks. but i doubt it should or will ever be in the docs. there are just too many factors that decide how fast things are for this to make sense in a place that should work as a definitive source of truth

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
this goes from CPUs to GPUs over motherboards, drives, ram, bios settings....operating system, kernel versions, drive settings and configurations, network card settings, drivers...... not even talking about the lotus software configuration itself... etc. etc. etc

**Reply by de-authority** [reply]
Yes, I know that there are too many factors that affect mining, but every small miner will start with hardware, so efficiency is something they will consider in the future. At the beginning, stability is the most important factor. So we just need hardware that meets the stability

**Reply by de-authority** [reply]
@f8-ptrk Most importantly, there are always small miners who ask me if I can recommend hardware to them

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
can you? 

i have a hard time with that. because i would build way more robust computers then actually needed. due to the fact that i have run a dozen of them and have seen most of the shit that can/could go wrong

**Reply by de-authority** [reply]
I can't now, It really needs a lot of hardware to test，the Chinese community has recommended hardware configuration, but they also run the modified master  code. I think it may not be suitable for all small miners
